From 974422b09c5c4dd868c54a2bef84d86b2e31e136 Mon Sep 17 00:00:00 2001
From: Colm O Heigeartaigh <coheigea@apache.org>
Date: Mon, 3 Sep 2012 12:41:42 +0000
Subject: [PATCH] [SANTUARIO-342] - NullPointer in
 javax.xml.crypto.KeySelectorException.printStackTrace

Conflicts:

	src/javax/xml/crypto/KeySelectorException.java

git-svn-id: https://svn.apache.org/repos/asf/santuario/xml-security-java/branches/1.4.x-fixes@1380221 13f79535-47bb-0310-9956-ffa450edef68
---
 CHANGELOG.txt                                  |  1 +
 src/javax/xml/crypto/KeySelectorException.java | 12 +++++++++---
 2 files changed, 10 insertions(+), 3 deletions(-)

diff --git a/CHANGELOG.txt b/CHANGELOG.txt
index 91f22a9887..3c884eaba6 100644
--- a/CHANGELOG.txt
+++ b/CHANGELOG.txt
@@ -1,6 +1,7 @@
 Changelog for "Apache xml-security" <http://santuario.apache.org/>
 
 New in v.1.4.8-SNAPSHOT:
+    Fixed SANTUARIO-342 - NullPointer in javax.xml.crypto.KeySelectorException.printStackTrace
     Fixed SANTUARIO-336 - Multiple race conditions in the ResolverDirectHttp implementation
     Fixed SANTUARIO-334 - UnsyncByteArrayOutputStream hangs on messages larger 512 MB.
 
diff --git a/src/javax/xml/crypto/KeySelectorException.java b/src/javax/xml/crypto/KeySelectorException.java
index 9ef6e27f01..cf18c3c90f 100644
--- a/src/javax/xml/crypto/KeySelectorException.java
+++ b/src/javax/xml/crypto/KeySelectorException.java
@@ -115,7 +115,9 @@ public Throwable getCause() {
      */
     public void printStackTrace() {
 	super.printStackTrace();
-	//XXX print backtrace of cause
+        if (cause != null) {
+            cause.printStackTrace();
+        }
     }
 
     /**
@@ -126,7 +128,9 @@ public void printStackTrace() {
      */
     public void printStackTrace(PrintStream s) {
 	super.printStackTrace(s);
-	//XXX print backtrace of cause
+        if (cause != null) {
+            cause.printStackTrace(s);
+        }
     }
 
     /**
@@ -137,6 +141,8 @@ public void printStackTrace(PrintStream s) {
      */
     public void printStackTrace(PrintWriter s) {
         super.printStackTrace(s);
-	//XXX print backtrace of cause
+        if (cause != null) {
+            cause.printStackTrace(s);
+        }
     }
 }
